app.js

```diff
@@ -135,19 +135,27 @@ const styles = StyleSheet.create({
   },
 })
 
+let settingsButtonActive = false
+
+function openSettings(route, navigator) {
+  navigator.push({
+    id: 'SettingsView',
+    title: 'Settings',
+    index: route.index + 1,
+    sceneConfig: Navigator.SceneConfigs.FloatFromBottom,
+  })
+}
 
 function LeftButton(route, navigator, index, navState) {
   switch (route.id) {
     case 'HomeView':
       return (
         <TouchableOpacity
           style={[styles.backButton]}
-          onPress={() => navigator.push({
-            id: 'SettingsView',
-            title: 'Settings',
-            index: route.index + 1,
-            sceneConfig: Navigator.SceneConfigs.FloatFromBottom,
-          })}
+          onPress={settingsButtonActive ? () => {} : () => {
```

Comment:
Import `noop` from lodash and use it here, and move the `settingsButtonActive = true` line into the `openSettings` function.

`onPress={settingsButtonActive ? noop : () => openSettings(route, navigator)}`